cachi2/core/package_managers/rpm/redhat.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import uuid
+from functools import cached_property
+from pathlib import Path
+from typing import Annotated, Any, Optional
+
+from pydantic import AfterValidator, BaseModel, PositiveInt, field_validator
+
+log = logging.getLogger(__name__)
+
+
+class Package(BaseModel):
+    """Package item; represents RPM or SRPM file."""
+
+    repoid: Optional[str] = None
+    url: str
+    checksum: Optional[str] = None
+    size: Optional[int] = None
+
+
+class Arch(BaseModel):
+    """Architecture structure."""
+
+    arch: str
+    packages: list[Package] = []
+    source: list[Package] = []
+
+
+def _arch_empty(arch: Arch) -> Arch:
+    """Validate arch."""
+    if arch.packages == [] and arch.source == []:
+        raise ValueError("At least one field ('packages', 'source') must be set in every arch.")
+    return arch
+
+
+MyArch = Annotated[Arch, AfterValidator(_arch_empty)]
```

Comment:
Is there a reason why we need this as a type validator instead of a model validator? IOW, this seems to work as expected and doesn't lead to this Annotated wildness:
```
    @model_validator(mode="after")
    def _arch_empty(self) -> "Arch":
        """Validate arch."""
        if self.packages == [] and self.source == []:
            raise ValueError("At least one field ('packages', 'source') must be set in every arch.")
        return self
```